client/python/gradio_client/client.py

```diff
@@ -247,7 +247,9 @@ def stream_messages(
                             resp = json.loads(line[5:])
                             if resp["msg"] == ServerMessage.heartbeat:
                                 continue
-                            elif resp["msg"] == ServerMessage.server_stopped:
+                            elif (
+                                resp.get("message", "") == ServerMessage.server_stopped
```

Comment:
When the server stops unexpectedly it sends an `unexpected_error` not a `server_stopped` message.

Reply:
Just to make sure I understand, if the server stops unexpectedly, `resp` will not have a `msg` key but instead have a `message` key? That seems weird, can we unify?

Reply:
Do we need to make any changes in the frontend, e.g. https://github.com/gradio-app/gradio/blob/bed2f82e2297b50f7b59423a3de05af0b9910724/client/js/src/helpers/api_info.ts#L210?

Reply:
It sends a response with `msg` and `message` keys but the `msg` key is `unexpected_error` as opposed to `server_stopped` so this if block never catches. I don't think it makes sense to check for `unexpected_error` in that codeblock because unexpected errors should not close the entire stream since they can be isolated to a single event id.